### PR TITLE
FEAT: Make a 3d second game

### PR DESCRIPTION
**Pull request by Thomaltarix**
I fully implement all the QuentinTheFarmerGame (In 2 days, so don't hope too much ^^) which is a T-Rex game like.

You can compile the new game by writing:
```bash
sudo cmake -B build -DQUENTIN_THE_FARMER_GAME=ON
```
To go back to the R-TypeGame, you should write this:
```bash
sudo cmake -B build -DQUENTIN_THE_FARMER_GAME=OFF
```

The player can jump but can't move.

For now, the jump isn't very beautiful due to a limited time to implement it.
There's only one tree that's coming towards the player, but since I updated the settings, only one tree is already really hard.

The tree comes towards the player again and again while the player isn't dead.

For now there isn't score system that save it into the database, but you can implement it if you want 😉 

The HitBoxes are shown for now, but you can remove it just by deleting this line:
image

### CODE REVIEW

**Review by Peralban — DISMISSED** [review]

File: Engine/Shared/DefaultEventHandlers/DrawHitBoxes/HitBoxesDrawHandler.cpp
Comment (mathieurobert1) [inline Engine/Shared/DefaultEventHandlers/DrawHitBoxes/HitBoxesDrawHandler.cpp:1]: All your indexes must be checked

File: Engine/Shared/DefaultEventHandlers/UpdateCamera/CameraUpdateHandler.cpp
Comment (mathieurobert1) [inline Engine/Shared/DefaultEventHandlers/UpdateCamera/CameraUpdateHandler.cpp:1]: Same here check indexes

File: Engine/Shared/DefaultEventHandlers/UpdateHitBox/HitBoxUpdateHandler.cpp
Comment (mathieurobert1) [inline Engine/Shared/DefaultEventHandlers/UpdateHitBox/HitBoxUpdateHandler.cpp:1]: Same

File: QuentinTheFarmerGame/EventHandlers/SinglePlayer/SinglePlayerHandler.cpp
Comment (mathieurobert1) [inline QuentinTheFarmerGame/EventHandlers/SinglePlayer/SinglePlayerHandler.cpp:1]: Same

**Review by mathieurobert1 — CHANGES_REQUESTED** [review]
Check indexes in events

File: Engine/Shared/DefaultEventHandlers/DrawHitBoxes/HitBoxesDrawHandler.cpp
Comment (Thomaltarix) [reply Engine/Shared/DefaultEventHandlers/DrawHitBoxes/HitBoxesDrawHandler.cpp:1]: Well seen

File: Engine/Shared/DefaultEventHandlers/UpdateCamera/CameraUpdateHandler.cpp
Comment (Thomaltarix) [reply Engine/Shared/DefaultEventHandlers/UpdateCamera/CameraUpdateHandler.cpp:1]: Done

File: Engine/Shared/DefaultEventHandlers/UpdateHitBox/HitBoxUpdateHandler.cpp
Comment (Thomaltarix) [reply Engine/Shared/DefaultEventHandlers/UpdateHitBox/HitBoxUpdateHandler.cpp:1]: Done

File: QuentinTheFarmerGame/EventHandlers/SinglePlayer/SinglePlayerHandler.cpp
Comment (Thomaltarix) [reply QuentinTheFarmerGame/EventHandlers/SinglePlayer/SinglePlayerHandler.cpp:1]: Done

**Review by Peralban — APPROVED** [review]

**Review by mathieurobert1 — APPROVED** [review]

**Pull request merged**